File: leet_BTREE_1379_find_in_cloned___1dequeBFS.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    TreeNode* getTargetCopy(TreeNode* o, TreeNode* c, TreeNode* t)
    {
        if (!o)
            return NULL;
        if (o == t)
            return c;

        deque<TreeNode*>    dq;

        dq.push_back(c);
        while (!dq.empty())
        {
            c = dq.front();
            dq.pop_front();
            if (c->val == t->val)
                return c;
            if (c->left)
                dq.push_back(c->left);
            if (c->right)
                dq.push_back(c->right);
        }
        return NULL;
    }
};

File: leet_BTREE_1379_find_in_cloned___1dequeBFS.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function getTargetCopy(original: TreeNode | null, cloned: TreeNode | null, target: TreeNode | null): TreeNode | null {
    if (!original)
        return null
    if (original == target)
        return cloned
    let dq = []
    dq.push(cloned)
    while (dq.length != 0) {
        cloned = dq.shift()
        if (cloned.val == target.val)
            return cloned
        if (cloned.left)
            dq.push(cloned.left)
        if (cloned.right)
            dq.push(cloned.right)
    }
    return null
};
